server.go

```diff
@@ -256,13 +267,14 @@ func handlePacket(s *Server, p orderedRequest) error {
 			rpkt = statusFromError(p.ID, err)
 		}
 	case *sshFxpOpendirPacket:
-		p.Path = toLocalPath(p.Path)
+		lp := toLocalPath(s.workDir, p.Path)
```

Comment:
There’s another `p.Path` that wasn’t replaced by `lp` in the `&sshFxpOpenPacket` packet. I’d probably recommend backing out the `s/lp/p.Path/` change instead.

Less change, less chance of new bug.

Reply:
I'm not sure I follow. The only other assignment to `p.Path` that I can find is in `(*sshFxpSetstatPacket).respond`.

This change was to prevent the following scenario: `toLocalPath(s.workDir, toLocalPath(s.workDir, p.Path))` which happens in `(*sshFxpOpenPacket).respond` because of this assignment to `p.Path`. This becomes a problem when workDir is being used and we're **not** checking `filepath.IsAbs`, resulting in double workDir prefix on Windows and Plan 9.

Reply:
Ach so! You’re absolutely right. The `toLocalPath` really should be deferred in that case until the `.respond` happens. Nice bug catch!